class/keycloak.yml

```
@@ -9,3 +14,57 @@ parameters:
           - keycloak/component/main.jsonnet
         input_type: jsonnet
         output_path: keycloak/
+      - output_path: keycloak/01_keycloak_helmchart
+        input_type: helm
+        output_type: yaml
+        input_paths:
+          - keycloak/helmcharts/keycloak
+        helm_values:
+          statefulsetLabels: ${keycloak:labels}
+          resources: ${keycloak:resources}
+          # extraEnv *MUST* be a string, as it's fed through a templating
+          # function.
+          extraEnv: |
+            - name: JAVA_OPTS
+              value: >-
+                        -XX:+UseContainerSupport
+                        -XX:MaxRAMPercentage=50.0
+                        -Djava.net.preferIPv4Stack=true
+                        -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS
+                        -Djava.awt.headless=true
+                        ${keycloak:extraJavaOpts}
+            - name: KEYCLOAK_STATISTICS
+              value: ${keycloak:monitoring:statistics}
```

You'll want to set the following env vars as well for Keycloak to work properly in K8s:
```suggestion
              value: ${keycloak:monitoring:statistics}
          - name: JGROUPS_DISCOVERY_PROTOCOL
            value: dns.DNS_PING
          - name: JGROUPS_DISCOVERY_PROPERTIES
            value: 'dns_query={{ include "keycloak.serviceDnsName" . }}'
          - name: CACHE_OWNERS_COUNT
            value: "${keycloak:replicas}"
          - name: CACHE_OWNERS_AUTH_SESSIONS_COUNT
            value: "${keycloak:replicas}"
          - name: PROXY_ADDRESS_FORWARDING
            value: "true"
```

And set `replicas` via a parameter.